[opt](systable) Improve information_schema.partitions getVisibleVersion latency under cloud mode

koarz opened the pull request:
Added a new `SessionVariable`: `cloud_partitions_table_use_cached_visible_version`. By executing `set cloud_partitions_table_use_cached_visible_version = true`, you can control whether to use the cached visible version when reading the `partitions` system table.

The `cloud_partitions_table_use_cached_visible_version` default value is true

Accessing the SessionVariable requires the ConnectContext. However, ConnectContext is thread-local. Since the RPC request from the BE creates a new thread (which does not contain the ConnectContext), we need to retrieve the correct ConnectContext using the original thread_id.

koarz committed:
[fix](systable)fix partitions table getVisiableVersion cost too many time

koarz commented:
run buildall

koarz committed:
fix

koarz commented:
run buildall

gavinchou commented on fe/fe-core/src/main/java/org/apache/doris/tablefunction/MetadataGenerator.java:
```diff
@@ -1679,7 +1679,8 @@ private static void partitionsForInternalCatalog(UserIdentity currentUserIdentit
                             .getBucketNum())); // BUCKET_NUM
                     trow.addToColumnValue(new TCell().setLongVal(partition.getCommittedVersion())); // COMMITTED_VERSION
-                    trow.addToColumnValue(new TCell().setLongVal(partition.getVisibleVersion())); // VISIBLE_VERSION
+                    trow.addToColumnValue(
```
add a session var or configuration to control the the behavior

koarz committed:
1

koarz committed:
2

koarz committed:
3

koarz committed:
Merge branch 'master' into fix-visiable-version

koarz commented on be/src/exec/schema_scanner/schema_partitions_scanner.cpp:
```diff
@@ -126,6 +126,9 @@ Status SchemaPartitionsScanner::get_onedb_info_from_fe(int64_t dbId) {
     schema_table_request_params.__set_current_user_ident(*_param->common_param->current_user_ident);
     schema_table_request_params.__set_catalog(*_param->common_param->catalog);
     schema_table_request_params.__set_dbId(dbId);
+    if (_param->common_param->thread_id > 0) {
```
Accessing the SessionVariable requires the ConnectContext. However, ConnectContext is thread-local. Since the RPC request from the BE creates a new thread (which does not contain the ConnectContext), we need to retrieve the correct ConnectContext using the original thread_id.

koarz committed:
1

hello-stephen commented:
# BE UT Coverage Report
Increment line coverage `0.00% (0/3)` :tada:

[Increment coverage report](http://coverage.selectdb-in.cc/coverage/d1c5e7e0da0ac35d1d6896345f2df9aa7e2caceb_d1c5e7e0da0ac35d1d6896345f2df9aa7e2caceb/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/d1c5e7e0da0ac35d1d6896345f2df9aa7e2caceb_d1c5e7e0da0ac35d1d6896345f2df9aa7e2caceb/report/index.html)
| Category          | Coverage           |
|-------------------|--------------------|
| Function Coverage | 53.52% (18736/35010) |
| Line Coverage     | 39.11% (172752/441676) |
| Region Coverage   | 33.75% (133868/396675) |
| Branch Coverage   | 34.68% (57585/166068) |

hello-stephen commented:
# FE Regression Coverage Report
Increment line coverage `  63.64% (7/11)` :tada:
[Increment coverage report](http://coverage.selectdb-in.cc/coverage/58515_d1c5e7e0da0ac35d1d6896345f2df9aa7e2caceb_merge_fe/increment_report/index.html)
[Complete coverage report](http://coverage.selectdb-in.cc/coverage/58515_d1c5e7e0da0ac35d1d6896345f2df9aa7e2caceb_merge_fe/report/index.html)

gavinchou committed:
Merge branch 'master' into fix-visiable-version

koarz committed:
Remove affectQueryResult from cloudPartitionsTableUseCachedVisibleVersion

Remove affectQueryResult attribute from cloudPartitionsTableUseCachedVisibleVersion variable.
